### ISSUE TITLE
Is this at all possible?

### ISSUE THREAD

**Issue by Barney2324** [Simbul/baker#1125]
If this is possible and anyone fancies giving me a quick guide, I would be most appreciative, is it possible or has anyone managed to successfully integrate fastpdfkit or any other PDF reader into Baker.

I don't want to upset anyone here and I know baker is HTML focused and is a HTML solution.

It's just I figured maybe someone may have an idea on how to get a PDF reader working on this project? Would make life so much easier for me.

**Comment by Barney2324** [comment]
I hate working with HTML5, I'm no webdev an so it doesn't come natural to me, laker looks good but it seems a lot harder to put a magazine together in HTML editor than in indesign or in my case, I use joomag which is a simple visual editor, I then download my PDF (it's not the best PDF creator but it's easy to use and they will iron out the bugs eventually)

I have purchased in5 and don't get on with it, for 2 reasons, 1) like others on here I found issues with viewing on iPhone and 2) I'm not currently designing much in Indesign, I gave up with it for now because it's a steep learning curve and I need the mag out quicker for first issue.

**Comment by Barney2324** [comment]
Ps I know plenty may not agree with my process but I haven't the time to put into learning indesign yet and I'm on a deadline.

**Comment by benoitchantre** [comment]
Baker is not intended to work with PDF for the main content. If you can make it work, you need to add some interactive elements, otherwise Apple will not approve your project as an app. Interactive elements (slideshows, videos, etc) in PDF are not supported on iOS.

**Comment by Barney2324** [comment]
Why does fastpdfkit add multimedia and interactive tutorials for their iOS PDF reader if it does not work?

I have seen a demo of their reader which has videos and interactive content...?

All I need to know is how to implement this into Baker succesfully.

**Comment by MaverickWar** [comment]
Yes I have seen this also, I too once toyed with trying to get PDF's to work with Baker.

If I I were you, I would give up on that, look back on here at any topic relating to PDF, it is either ignored or shot down in flames, Baker will not give you any support to do this. I guess it's because they want to push HTML5 so hard...

I'd be very surprised if they help you find a way to plug this into baker. Would be nice if they did though, it looks like a great reader and I would use it if they did. Just don't get your hopes up.

**Comment by benoitchantre** [comment]
There's no tutorial to implement this library in Baker.
You can find the documentation on the website of FastPdfKit, but you need to understand Objective-C and how Baker is build to make it work.

**Comment by Barney2324** [comment]
That's why I have asked here to see if someone might please be able to help me, I would be very grateful. My knowledge of baker is basic and the documentation from fastpdfkit.com isn't very comprehensive.

Also as you said, baker has no documentation at all on how to change the default reader.

If anyone could help, it would be greatly appreciated.

**Comment by folletto** [comment]
_"I guess it's because they want to push HTML5 so hard"_ — What? And where this is coming from? :D

I'm actually a big fan of PDF as a standard. It's just not what Baker is for, because it has never been designed to support PDF, work with PDF or anything related to it, and will more than duplicate the work to make it work. If you want to do it fine, you can branch it freely. As much it might look simple to say "Oh, I can just change the reader" that is pretty much like re-writing everything from scratch. You can do that, sure, but that's not Baker anymore.

tl;dr — There's no point in using Baker if you need to work with PDF! :)

**Comment by Barney2324** [comment]
I thought a new reader could be simply plugged into baker and enabled?

**Comment by Barney2324** [comment]
Also why has this topic been closed when it has not been really looked at...

**Comment by benoitchantre** [comment]
Please read again the answer of @folletto.

**Comment by Barney2324** [comment]
I have read it but I don't see any help implementing unless I'm going blind? All I see is 'this has nothing to do with us' and 'topic closed' have I missed some help offered somewhere here?

**Comment by benoitchantre** [comment]
@Barney2324 You can't be helped if what you ask about a total rework of the Baker in the opposite direction to the one that was choosen for this project.

> As much it might look simple to say "Oh, I can just change the reader" that is pretty much like re-writing everything from scratch. You can do that, sure, but that's not Baker anymore.
